src/documentation/chunk.rs

```diff
@@ -107,80 +107,92 @@ impl CheckableChunk {
             &range
         );
 
-        let Range { start, end } = range.clone();
+        let Range { start, end } = range;
         let mut active = false;
         self.source_mapping
             .iter()
-            .skip_while(|(sub, _span)| sub.end <= start)
-            .take_while(|(sub, _span)| end <= sub.end)
+            .skip_while(|(fragment_range, _span)| fragment_range.end <= start)
+            .take_while(|(fragment_range, _span)| end <= fragment_range.end)
             .inspect(|x| {
                 trace!(">>> item {:?} ∈ {:?}", &range, x.0);
             })
-            .filter(|(sub, _)| {
+            .filter(|(fragment_range, _)| {
                 // could possibly happen on empty documentation lines with `///`
-                sub.len() > 0
+                fragment_range.len() > 0
             })
-            .filter_map(|(sub, span)| {
-                if span.start.line == span.end.line {
-                    debug_assert!(span.start.column <= span.end.column);
-                    if span.start.column > span.end.column {
-                        return None;
-                    }
+            .filter_map(|(fragment_range, fragment_span)| {
+                // trim range so we only capture the relevant part
+                let sub_fragment_range = std::cmp::max(fragment_range.start, range.start)
+                    ..std::cmp::min(fragment_range.end, range.end);
+                trace!(
+                    ">> fragment: span:{:?} => range:{:?} | sub:{:?} -> sub_fragment{:?}",
+                    &fragment_span,
+                    &fragment_range,
+                    range,
+                    &sub_fragment_range,
+                );
+
+                if sub_fragment_range.len() == 0 {
+                    log::debug!("sub fragment is zero, dropping!");
+                    return None;
                 }
-
-                // assured by filter
-                assert!(end > 0);
-
-                if sub.contains(&start) {
-                    let offset = start - sub.start;
-                    if sub.contains(&(end - 1)) {
-                        // complete start to end
-                        active = false;
-                        Some((start..end, offset))
-                    } else {
-                        // only start, continue taking until end
-                        active = true;
-                        Some((start..sub.end, offset))
+                if fragment_span.start.line == fragment_span.end.line {
+                    assert_eq!(
+                        fragment_span.end.column - fragment_span.start.column + 1,
+                        fragment_range.len()
+                    );
+                } else {
+                    assert!(fragment_span.start.column <= fragment_span.end.column);
```

Comment:
Why is the `end.column` always smaller than then `start.column`? In other words, why is the number of characters in the first line always higher than the number of characters in the following lines?

Reply:
Just fixed that one :>